RecoTauTag/HLTProducers/src/PFJetsTauOverlapRemoval.cc

```diff
@@ -0,0 +1,73 @@
+#include "RecoTauTag/HLTProducers/interface/PFJetsTauOverlapRemoval.h"
+#include "Math/GenVector/VectorUtil.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "DataFormats/TauReco/interface/PFTau.h"
+
+//
+// class declaration
+//
+using namespace reco   ;
+using namespace std    ;
+using namespace edm    ;
+using namespace trigger;
+
+PFJetsTauOverlapRemoval::PFJetsTauOverlapRemoval(const edm::ParameterSet& iConfig):
+  tauSrc    ( consumes<trigger::TriggerFilterObjectWithRefs>(iConfig.getParameter<InputTag>("TauSrc"      ) ) ),
+  PFJetSrc( consumes<PFJetCollection>(iConfig.getParameter<InputTag>("PFJetSrc") ) )
+{  
+  produces<PFJetCollection>();
+}
+PFJetsTauOverlapRemoval::~PFJetsTauOverlapRemoval(){ }
+
+void PFJetsTauOverlapRemoval::produce(edm::StreamID iSId, edm::Event& iEvent, const edm::EventSetup& iES) const
+{
+    
+  unique_ptr<PFJetCollection> cleanedPFJets(new PFJetCollection);
+    
+  double deltaR    = 1.0;
+  double matchingR = 0.5;
+  
+  edm::Handle<trigger::TriggerFilterObjectWithRefs> tauJets;
+  iEvent.getByToken( tauSrc, tauJets );
+  
+  edm::Handle<PFJetCollection> PFJets;
+  iEvent.getByToken(PFJetSrc,PFJets);
+                
+  trigger::VRpftau taus; 
+  tauJets->getObjects(trigger::TriggerTau,taus);
+
+  if(PFJets->size() == 2 || PFJets->size() > 3){
+    for(unsigned int iTau = 0; iTau < taus.size(); iTau++){  
+      for(unsigned int iJet = 0; iJet < PFJets->size(); iJet++){
+        const PFJet &  myPFJet = (*PFJets)[iJet];
+        deltaR = ROOT::Math::VectorUtil::DeltaR((taus[iTau])->p4().Vect(), myPFJet.p4().Vect());
+        if(deltaR > matchingR) cleanedPFJets->push_back(myPFJet);
+        break;
+      }
+    }
+  }
+  else if(PFJets->size() == 3){
+    for(unsigned int iTau = 0; iTau < taus.size(); iTau++){  
```

Comment:
hi Martin,

the general answer is this module is specific to the VBF seed and its corresponding HLT path (in development). The seed is such that it can have either 2 or 3 jets, and in case it selects 3 jets, we want to clean only those two that make up the larger m_jj.

I let Albert comment on the specific implementation though.